homeassistant/components/twitch/sensor.py

```diff
@@ -94,13 +94,20 @@ async def async_setup_entry(
     """Initialize entries."""
     client = hass.data[DOMAIN][entry.entry_id]
 
-    async_add_entities(
-        [
-            TwitchSensor(channel, client)
-            async for channel in client.get_users(logins=entry.options[CONF_CHANNELS])
-        ],
-        True,
-    )
+    channels = entry.options[CONF_CHANNELS]
+
+    entities: list[TwitchSensor] = []
+
+    # Split channels into chunks of 100 to avoid hitting the rate limit
+    for chunk in (channels[i : i + 100] for i in range(0, len(channels), 100)):
```

Comment:
I remember I had a similar problem with `steam_online` where I created a function to return an iterator. 

https://github.com/home-assistant/core/blob/8fd0a1b0836b538295f987a284a44322bf822376/homeassistant/components/steam_online/config_flow.py#L114

Would it make sensor for HA to provide this as an util function?

Reply:
#101226

Reply:
Done